rdma_cli/src/egress.rs

```diff
@@ -322,12 +319,16 @@ fn wait(epoll_fd: i32, gatewayCli: &GatewayClient, fds: &mut HashMap<i32, FdType
                                 }
                                 RDMARespMsg::RDMANotify(response) => {
                                     if response.event & EVENT_IN != 0 {
-                                        let mut channelToSockInfos =
-                                            gatewayCli.channelToSockInfos.lock();
-                                        let sockInfo = channelToSockInfos
-                                            .get_mut(&response.channelId)
-                                            .unwrap();
-                                        gatewayCli.WriteToSocket(sockInfo, &sockFdMappings);
+                                        let mut sockInfo;
+                                        {
+                                            let mut channelToSockInfos =
+                                                gatewayCli.channelToSockInfos.lock();
+                                            sockInfo = channelToSockInfos
+                                                .get_mut(&response.channelId)
+                                                .unwrap()
+                                                .clone();
+                                        }
```

Comment:
better to consolidate this section of code since it's duplicated in a couple of locations.

Reply:
Fixed